### ISSUE TITLE
`nuxi-edge@latest upgrade` ERROR  Invalid regular expression flag 

### ISSUE THREAD

**Issue by cawa-93** [nuxt/framework#6749]
### Environment

- Operating System: `Linux`
- Node Version:     `v16.14.2`
- Nuxt Version:     `3.0.0-rc.8`
- Package Manager:  `npm@7.17.0`
- Builder:          `vite`
- User Config:      `-`
- Runtime Modules:  `-`
- Build Modules:    `-`

### Reproduction

1. Open https://stackblitz.com/edit/github-hxvhp5?file=package.json
2. run `npm i && npx nuxi-edge@latest upgrade -h`

### Describe the bug

```
❯ npx nuxi-edge@latest upgrade -h
Nuxt CLI v3.0.0-rc.8-27680175.faf00f5                                                                        17:13:26

 ERROR  Invalid regular expression flag (35:17)                                                              17:13:26

  at _0x56d924 (https://github-hxvhp5.w-corp.staticblitz.com/blitz.8d192f41f47e98afc85266040b8c76ffdc5aedf4.js:15:288679)
  at <anonymous> (https://github-hxvhp5.w-corp.staticblitz.com/blitz.8d192f41f47e98afc85266040b8c76ffdc5aedf4.js:6:982310)
  at _createModuleJob2/s< (https://github-hxvhp5.w-corp.staticblitz.com/blitz.8d192f41f47e98afc85266040b8c76ffdc5aedf4.js:6:1211628)
  at <anonymous> (<anonymous>)
```

**Comment by danielroe** [comment]
Reproducible when built in monorepo as well. Generated code is:

```js
/*---------------------------------------------------------------------------------------------
 *  Copyright (c) Microsoft Corporation. All rights reserved.
 *  Licensed under the MIT License. See License.txt in the project root for license information.
 *--------------------------------------------------------------------------------------------*/
var __assign = (/framework/packages/nuxi && /framework/packages/nuxi.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
```

I think we've encountered this before as well in another issue which we resolved simply by not inlining the dependency. But honestly this seems like it might be a rollup bug. Any ideas @pi0?

**Comment by pi0** [comment]
Seems regression of https://github.com/nuxt/framework/pull/6707 that adds using `pkg-types` > `jsonc-parser` 

But definitely a rollup bug since first it had to the tree-shake the library (sub-dep is for ts parsing) and second it is replacing without escpaing.

**Comment by pi0** [comment]
Thanks for reporting. upgrade command should be fixed in latest nuxi edge (try with v3.0.0-rc.8-27681070.ad63817)
